js_modules/dagit/packages/core/src/assets/AssetsCatalogTable.tsx

```diff
@@ -203,6 +226,68 @@ export const AssetsCatalogTable: React.FC<AssetCatalogTableProps> = ({
   );
 };
 
+const AssetGroupSuggest: React.FC<{
+  assets: Asset[];
+  value: AssetGroupSelector | null;
+  onChange: (g: AssetGroupSelector | null) => void;
+}> = ({assets, value, onChange}) => {
+  const assetGroups = React.useMemo(
+    () =>
+      uniqBy(
+        (assets || []).map(buildAssetGroupSelector).filter((a) => !!a) as AssetGroupSelector[],
+        (a) => JSON.stringify(a),
+      ).sort((a, b) => a.groupName.localeCompare(b.groupName)),
+    [assets],
+  );
+
+  const repoContextNeeded = React.useMemo(() => {
+    const result: {[groupName: string]: boolean} = {};
+    assetGroups.forEach(
+      (group) => (result[group.groupName] = group.groupName in result ? true : false),
```

Comment:
`result.hasOwnProperty(group.groupName)` just in case there's a `groupName` that matches a property in the prototype chain?

Reply:
Mm good call, can't be too safe.. I have a bad tendency of using `in` in all these scenarios...